src/mpid/ch4/shm/posix/eager/fbox/fbox_recv.h

```diff
@@ -113,14 +112,13 @@ MPIDI_POSIX_eager_recv_begin(MPIDI_POSIX_eager_recv_transaction_t * transaction)
 #endif /* POSIX_FBOX_DEBUG */
 
             /* We found a message so return success and stop. */
-            mpi_errno = MPIDI_POSIX_OK;
-            goto fn_exit;
+            MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPIDI_POSIX_FBOX_EAGER_RECV_BEGIN);
+            return MPIDI_POSIX_OK;
```

Comment:
How can I reproduce the warning here? I tried a strict build with gcc 4.8.5 but didn't see anything. I'm skeptical that the compiler would differentiate this new code from the previous. Additionally, this pattern of multiple `MPIR_FUNC_VERBOSE_EXIT + return` is not common in the codebase now and I'm not sure we should introduce it.

Reply:
This warning is indeed tricky. It seems that Linux developers found `-Wmaybe-uninitialized` to be so unreliable prior to GCC 4.9 that they just disabled it. https://patchwork.kernel.org/patch/9515907/#20232915

Reply:
If you prefer, I could filter this warning for `gcc-4`.

Reply:
I think Ken was simply asking how you found the warning, i.e., what compiler settings did you try for the warning to show up?

Reply:
@pavanbalaji This is produced with `--enable-strict --enable-g=all`. Ken has reproduced this warning already.